api/src/server.rs

```diff
@@ -191,6 +192,8 @@ pub trait EventStore: Send + Sync {
         limit: i64,
     ) -> Result<(i64, Vec<Cid>)>;
 
+    async fn highwater_mark(&self) -> Result<i64>;
```

Comment:
Hm. I was going to make the change, but this is defining behavior for the store. I kind of do like the difference in names between crates since it's a highwater mark from the store's perspective, and the fact that the API chooses to use that as a resume token is an implementation detail... I do find it obnoxious to call it two things in one crate though so I'm willing to make the change. Let me know what you think.

More generally, making this change made me think having these traits is kind of silly. I'd prefer to use the hyper server context and just pull out our ApiContext struct that has a database pool and whatever things we need to make a query without having to update multiple traits/files.

Reply:
I don't feel super strongly either way.  The discrepancy in terms jumped out at me, but I'm happy to defer to whatever you think makes the most sense.